content/general/authentication.textile

```diff
@@ -139,22 +139,91 @@ ably.requestToken({ clientId: 'client@example.com' }, function(err, token) {
+  }
+  var claims = {
+    "exp": currentTime + 3600
+  }
+  var base64Header = btoa(header);
+  var base64Claims = btoa(claims);
+  /* Apply the hash specified in the header */
+  var signature = hash((base64Header + "." + base64Claims), SECRET);
+  var jwt = base64Header + "." + base64Claims + "." + signature;
+  /* Send jwt to client */
+});
+
+*Note:* The authenticity of the JWT *will not* be checked, due to Ably not having access to your SECRET key.
+
+h2(#capabilities-explained). Capabilities and Token Security explained
+
+"API keys":https://support.ably.io/solution/articles/3000030054-what-is-an-app-api-key, like "Ably-compatible tokens":#tokens, have a set of capabilities assigned to them that specify which "operations":#capability-operations (such as subscribe or publish) can be performed on which channels. However, unlike tokens, API keys are long-lived, secret and typically not shared with un-trusted clients.
+
+API keys and their capabilities are "configured using the dashboard":https://support.ably.io/support/solutions/articles/3000030502-setting-up-and-managing-api-keys, they cannot be added or removed programmatically. Ably-compatible tokens on the other hand are designed to be shared with un-trusted clients, are short-lived, and significantly, they are configured and issued programmatically. See "selecting an authentication scheme":#selecting-auth to understand why token authentication, in most cases, is the preferred authentication scheme.
+
+"Ably Tokens":#tokens are issued from an existing "API key":https://support.ably.io/solution/articles/3000030054-what-is-an-app-api-key, and their capabilities can, at most, match the capabilities of the issuing API key. "Ably JWTs":#ably-jwt have a similar restriction on capabilities, correlating to the API key they've been signed with. If an API key must be shared with a third party, then it is recommended that "the principle of least privilege":http://en.wikipedia.org/wiki/Principle_of_least_privilege is considered, assigning only the capabilities needed by that third party. Thus, any Ably requests authenticated using that API key or Ably-compatible tokens associated with that API key, will be restricted to the capabilities assigned to the API key.
+
+Capabilities when requesting Ably Tokens are determined as follows:
 
+* If no capability is specified in the Ably Token request, then the Ably Token will be given the full set of capabilities assigned to the issuing key, "see example":#capabilities-explained-all;
+* If a set of capabilities are requested, then the Ably Token will be assigned the intersection of the requested capability and the capability of the issuing key, "see example":#capabilities-explained-intersection;
+* If a set of capabilities are requested, and the intersection between those and the API key's capabilities is empty (i.e. they are entirely incompatible), then the Ably Token request will result in an error, "see example":#capabilities-explained-error.
```

Comment:
While we're on the subject of grammar I think we should be using open punctuation, so for example we would have "ie" instead of "i.e.".

See for example https://www.cam.ac.uk/brand-resources/guidelines/editorial-style-guide for a standard.